### DISCUSSION TITLE
part-ii/2009-26/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#153] [Question comments]
Part II 2009 Coding and Cryptography: Paper 1, Section I, H 

https://questions.tripos.org/part-ii/2009-26/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

(i) Our method corresponds to finding an optimal binary code with these probabilities. Specifically, the $i$th character of the $j$th codeword tells us whether or not we should include the $j$th bulb in the $i$th trial (we take $1$ to mean inclusion and $0$ to mean exclusion). If this code is decipherable, then the defective bulb will be uniquely determined by these trials (specifically, the defective bulb is the one where no bulbs light up exactly when that bulb is used in the trial). If some codewords are shorter than others, it doesn't matter whether we include the bulbs from the shorter codewords in later tests (if we have done all the tests from the codeword and still do not know which bulb is defective, we know that that bulb is not defective).

We get the following results:

$$
10/55 = 11 \\
9/55 = 000 \\
8/55 = 010 \\
7/55 = 011 \\
6/55 = 100 \\
5/55 = 0010 \\
4/55 = 0011 \\
3/55 = 1010 \\
2/55 = 10110 \\
1/55 = 10111 \\
$$

(ii) Yes, such a procedure exists. One way to obtain one is to swap $0$s and $1$s in the above code; this preserves the expected word length because all words stay the same length.

(iii) Yes, a procedure that gives a smaller $N$ exists. The procedure in part (i) gives $N=6$, but we can achieve $N=4$ by using binary expansion; take

$$
10/55 = 1010 \\
9/55 = 1001 \\
8/55 = 1000 \\
7/55 = 0111 \\
6/55 = 0110 \\
5/55 = 0110 \\
4/55 = 0100 \\
3/55 = 0011 \\
2/55 = 0010 \\
1/55 = 0001 \\
$$

</details>
